ephios/static/ephios/js/event_copy.js

```diff
@@ -1,27 +1,145 @@
+                        break
+                    case frequencies.MONTHLY:
+                        isValid = isValid && (rule.month_mode === "bymonthday" && rule.bymonthday || rule.month_mode === "bysetpos" && rule.bysetpos && rule.byweekday)
+                        break
+                    case frequencies.YEARLY:
+                        isValid = isValid && (rule.year_mode === "bymonthday" && rule.bymonthday && rule.bymonth || rule.year_mode === "bysetpos" && rule.bysetpos && rule.byweekday && rule.bymonth)
+                        break
                 }
-                $('#rrule_occurrences').html("<p>" + gettext("No dates selected") + "</p>");
+
+                // end date set
+                isValid = isValid && (rule.end_mode === "COUNT" && rule.count || rule.end_mode === "UNTIL" && rule.until)
+
+                return isValid
             }
-        });
-    });
-});
 
-function parseIsoDatetime(dtstr) {
-    var dt = dtstr.split(/[: T-]/).map(parseFloat);
-    return new Date(dt[0], dt[1] - 1, dt[2], dt[3] || 0, dt[4] || 0, dt[5] || 0, 0);
-}
+            const rrule_set = computed(() => {
+                let set = new rrule.RRuleSet()
+                rules.value.forEach(rule => {
+                    if (!isRuleValid(rule)) {
+                        return
+                    }
+                    set.rrule(new rrule.RRule({
+                        freq: rule.freq,
+                        interval: rule.interval,
+                        dtstart: rrule.datetime(...DTSTART.value.split("-"), ...formatHourOrZero(DTSTART_TIME.value, pickHour)),
+                        byweekday: rule.byweekday,
+                        bymonthday: rule.month_mode === "bymonthday" ? rule.bymonthday : undefined,
+                        bysetpos: rule.month_mode === "bysetpos" ? rule.bysetpos : undefined,
+                        bymonth: rule.bymonth,
+                        count: rule.end_mode === "COUNT" ? rule.count : undefined,
+                        until: rule.end_mode === "UNTIL" && rule.until ? rrule.datetime(...rule.until.split("-"), formatHourOrZero(rule.UNTIL_TIME, pickHour)) : undefined,
+                        tzid: Intl.DateTimeFormat().resolvedOptions().timeZone,
+                    }))
+                })
+                dates.value.forEach(date => {
+                    if (date.date && (!pickHour || date.time)) {
+                        set.rdate(rrule.datetime(...date.date.split("-"), ...formatHourOrZero(date.time, pickHour)))
+                    }
+                })
+                return set
+            })
+
+            const computed_dates = computed(() => {
+                return rrule_set ? rrule_set.value.all().map(date => {
+                    return formatDate(date) + " " + (date.getUTCHours() < 10 ? "0" : "") + date.getUTCHours() + ":" + (date.getUTCMinutes() < 10 ? "0" : "") + date.getUTCMinutes()
+                }): []
+            })
```

Comment:
can you factor out adding the leading zeros?

Reply:
why and to where?

Reply:
because this is hard to read